AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1515,40 +1515,36 @@ - (void)_removeSubnode:(ASDisplayNode *)subnode
   [subnode __setSupernode:nil];
 }
 
+// NOTE: You must not called this method while holding the receiver's property lock. This may cause deadlocks.
```

Comment:
Someone correctly mentioned that we need a more comprehensive locking refactor. Comments like this bear that out.

I've manually verified that this condition is currently satisfied. Unfortunately I don't know of any way to assert that the current thread is not holding a given lock, so we don't have great enforcement options here =/